src/components/NurseryWithdrawals/PlantingProgressMap.tsx

```diff
@@ -38,16 +38,20 @@ export default function PlantingProgressMap({ plantingSiteId }: PlantingProgress
   const [dispatching, setDispatching] = useState(false);
   const [requestId, setRequestId] = useState<string>('');
   const updateStatus = useAppSelector((state) => selectUpdatePlantingCompleted(state, requestId));
+  const [focusEntities, setFocusEntities] = useState<{ sourceId: string; id: number }[]>([]);
   const snackbar = useSnackbar();
 
   useEffect(() => {
-    if (plantingSite?.boundary) {
-      setMapData(MapService.getMapDataFromPlantingSite(plantingSite));
-    } else {
-      setMapData(undefined);
+    if (!mapData?.site?.entities || plantingSite?.id !== mapData.site.entities[0]?.id) {
+      if (plantingSite?.boundary) {
+        setMapData(MapService.getMapDataFromPlantingSite(plantingSite));
+        setFocusEntities([{ sourceId: 'sites', id: plantingSite?.id }]);
+      } else {
+        setMapData(undefined);
```

Comment:
clear focus entities just for completeness?

Reply:
sure that probably makes sense